### DISCUSSION TITLE
Is there a way to pre-populate form fields with dynamic values?

### DISCUSSION THREAD

**Discussion by fuzl-llc** [OrchardCMS/OrchardCore#9706] [Ideas]
I want to create a form field generated by OrchardCore.Forms with a dynamic value pre-populated e.g. a hidden fields that tells me what area of the site the user was in when they submitted the form (I'm going to have the form at the bottom of multiple pages on the site). I thought since liquid is so pervasive in the system I'd be able to use liquid to put some value into a form field's "default value" but it seems I cannot. Is there some other way to do this? Has adding liquid support been discussed for form fields? If not, any reason this shouldn't go in the backlog?

**Comment by ns8482e** [comment]
Add an `InputPart` with Type `Hidden` and provide `DefaultValue`.
Use javascript to update the value if needed based on `window.location`.
